flex-checks/src/main/java/org/sonar/flex/checks/FileHeaderCheck.java

```diff
@@ -60,75 +54,57 @@
     defaultValue = "false")
   public boolean isRegularExpression = false;
 
-  private Charset charset;
-  private String[] expectedLines;
+  private Predicate<String> headerMatcher;
 
-  private Pattern regularExpression = null;
-
-  @Override
-  public void setCharset(Charset charset) {
-    this.charset = charset;
-  }
-
-  @Override
-  public void init() {
-    if (isRegularExpression) {
-      if (headerFormat.endsWith("\n") || headerFormat.endsWith("\r")) {
-        regularExpression = Pattern.compile(headerFormat);
+  public Predicate<String> headerMatcher() {
+    if (headerMatcher == null) {
+      if (isRegularExpression) {
+        String regexp = headerFormat;
+        if (!headerFormat.endsWith("\n") && !headerFormat.endsWith("\r")) {
+          regexp += "(\r|\r\n|\n)";
+        }
+        Pattern pattern = Pattern.compile(regexp);
+        headerMatcher = s -> matchesRegularExpression(pattern, s);
       } else {
-        regularExpression = Pattern.compile(headerFormat + "(\r|\r\n|\n)");
+        String[] expectedLines = headerFormat.split("(?:\r)?\n|\r");
+        headerMatcher = s -> matchesPlainTextHeader(expectedLines, s);
       }
-    } else {
-      expectedLines = headerFormat.split("(?:\r)?\n|\r");
     }
+    return headerMatcher;
+  }
+
+  @Override
+  public List<AstNodeType> subscribedTo() {
+    return Collections.emptyList();
   }
 
   @Override
   public void visitFile(AstNode astNode) {
-    File file = getContext().getFile();
-    boolean hasHeader = isRegularExpression ? matchesRegularExpression(file) : matchesPlainTextHeader(file);
+    String fileContent = getContext().fileContent();
+    boolean hasHeader = headerMatcher().test(fileContent);
     if (!hasHeader) {
-      getContext().createFileViolation(this, "Add or update the header of this file.");
+      addFileIssue("Add or update the header of this file.");
     }
   }
 
-  @VisibleForTesting
-  protected boolean matchesRegularExpression(File file) {
-    String content;
-    try {
-      content = Files.toString(file, charset);
-    } catch (IOException e) {
-      throw new IllegalStateException(exceptionMessage(file), e);
-    }
+
+  private boolean matchesRegularExpression(Pattern regularExpression, String content) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "matchesRegularExpression" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.flex%3Aflex&branch=SONARFLEX-105&issues=AWGPc9zyOl9lQhPgXcMt&open=AWGPc9zyOl9lQhPgXcMt)

[ISSUE_KEY]: AWGPc9zyOl9lQhPgXcMt